### ISSUE TITLE
Low Precision while training

### ISSUE THREAD

**Issue by lijo381** [Deci-AI/super-gradients#1146]
Training a yolonas model on satellite image dataset, the mAP values and recall improve as expected. But the precision is very low because of which inference is very poor. Loss used is `ppyoloeloss` with SGD optimizer which internally uses focal loss, but looks like it is not helping the precision. Is there any way to improve the precision from training side ?

**Comment by BloodAxe** [comment]
Can you provide more information about your experiment?
1) Do you use pre-trained weights of yolo-nas?
2) Can you share the train/validation loss & mAP plots?
3) What are the size of objects in your dataset? Maybe they are too small?
4) Does the same behaviour holds when using Adam?

**Comment by lijo381** [comment]
>Do you use pre-trained weights of yolo-nas?
yes i used the coco weights
> Can you share the train/validation loss & mAP plots?
Shared as attachments below
>What are the size of objects in your dataset? Maybe they are too small?
The objects are pretty small as well there is imbalance in dataset. My assumption was focal loss will take care of imbalance.

>Does the same behaviour holds when using Adam?
Yes the behaviour is same with adam, though with SGD i am seeing faster convergence.

![Screenshot 2023-06-08 at 10 34 55 AM](https://github.com/Deci-AI/super-gradients/assets/13690209/6b9bac7c-1236-4ca7-b6c6-c5018fce8329)
![Screenshot 2023-06-08 at 10 36 08 AM](https://github.com/Deci-AI/super-gradients/assets/13690209/fa6f0fa9-5506-445b-a37c-a876a93aaf77)
![Screenshot 2023-06-08 at 10 36 37 AM](https://github.com/Deci-AI/super-gradients/assets/13690209/37a09b75-f399-4731-8ce8-c11c08c99013)
![Screenshot 2023-06-08 at 10 36 56 AM](https://github.com/Deci-AI/super-gradients/assets/13690209/b60a5d5e-55fd-4f65-8e73-a112f95ab388)
![Screenshot 2023-06-08 at 10 37 07 AM](https://github.com/Deci-AI/super-gradients/assets/13690209/83424b72-242b-4aee-a6a4-ca85a1ea645d)

**Comment by BloodAxe** [comment]
This clearly looks like an overfit.
Neither focal loss nor certain architecture can solve this magically.
You can also try adding DetectionMixup augmentation which usually helps.
 
On RF-100 we observe this is important not to filter any boxes even small once: https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/recipes/dataset_params/roboflow_detection_dataset_params.yaml#L23
You may also want to use this trick in your training.

**Comment by lijo381** [comment]
> You can also try adding DetectionMixup augmentation which usually helps.

Does this needs to be implemented seperately or is just a case of enabling ? If its the latter could you please point to it.

> Neither focal loss nor certain architecture can solve this magically.

I was able to generalise the yolov5 model for the same dataset with focal loss.

> not to filter any boxes even small once

will give it a try

**Comment by ShpihanVlad** [comment]
Hi, @lijo381 , @utkutpcgl , were your attempts successfull? Thx for your time

**Comment by utkutpcgl** [comment]
Not really, I had better results with yolov5, I might be missing something.

**Comment by ShpihanVlad** [comment]
Hi @utkutpcgl , in case your mAP scores were good enough, but precision is so small, #1305 may help you. For comparison, YOLOv5 calculates P and R on optimal confidence threshold, found by F1-curve

**Comment by BloodAxe** [comment]
Since 3.6 release we have enabled optimal confidence threshold computation by default in our DetectionMetrics so it should be higher now.
